catroid/src/androidTest/java/org/catrobat/catroid/test/formulaeditor/FormulaElementTest.java

```diff
@@ -129,6 +129,23 @@ public void testContainsElement() {
 				!formulaElement.containsElement(ElementType.NUMBER));
 	}
 
+	public void testInterpretOperatorEqual() {
+		FormulaElement formulaElement = new FormulaElement(ElementType.OPERATOR, Operators.EQUAL.name(), null,
+				new FormulaElement(ElementType.NUMBER, Integer.toString(-0), null), new FormulaElement(ElementType
+				.NUMBER, Integer.toString(0), null));
+		assertEquals("interpretOperatorEqual: -0 should be equal to 0", 1d, formulaElement.interpretRecursive(null));
+
+		formulaElement = new FormulaElement(ElementType.OPERATOR, Operators.EQUAL.name(), null,
+				new FormulaElement(ElementType.NUMBER, "NaN", null), new FormulaElement(ElementType.NUMBER, "NaN",
+				null));
+		assertEquals("interpretOperatorEqual: NaN should be equal to NaN", 1d, formulaElement.interpretRecursive(null));
+
+		formulaElement = new FormulaElement(ElementType.OPERATOR, Operators.EQUAL.name(), null,
+				new FormulaElement(ElementType.NUMBER, Integer.toString(0), null), new FormulaElement(ElementType.NUMBER,
+				Integer.toString(-0), null));
+		assertEquals("interpretOperatorEqual: 0 should be equal to -0", 1d, formulaElement.interpretRecursive(null));
+	}
```

Comment:
To me, this class is the right place to test operators!
However, all operator tests are currently located at [ParserTestOperators.java](https://github.com/Catrobat/Catroid/blob/f471d0c3abba61eaf03f56938909b7ea6219d5ca/catroid/src/androidTest/java/org/catrobat/catroid/test/formulaeditor/ParserTestOperators.java#L142). So for now it's better to move this test to the [ParserTestOperators](https://github.com/Catrobat/Catroid/blob/f471d0c3abba61eaf03f56938909b7ea6219d5ca/catroid/src/androidTest/java/org/catrobat/catroid/test/formulaeditor/ParserTestOperators.java) and adapt it accordingly. Just to be consistent. You can also increase the quality by splitting the test into e.g. `testNegativeZeroEqualsZero` (a better name!) and `testEqualsNaN`*.

\* The _NaN_ has nothing to do with the _negative zero_ test. I created a new Jira Ticket for that (https://jira.catrob.at/browse/CAT-2396). Should've done that in the first place :flushed: Would be awesome if you could handle that first since you already did it here. :smile_cat: